frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts

```diff
@@ -255,5 +396,35 @@ export class ConsoleRemotePlugin implements webpack.WebpackPluginInstance {
         }
       });
     }
+
+    const transformImports =
+      sharedDynamicModuleSettings.transformImports ??
+      ((moduleRequest) =>
+        !moduleRequest.split(path.sep).includes('node_modules') &&
+        /\.(jsx?|tsx?)$/.test(moduleRequest));
+
+    compiler.hooks.thisCompilation.tap(ConsoleRemotePlugin.name, (compilation) => {
+      const modifiedModules: string[] = [];
+
+      webpack.NormalModule.getCompilationHooks(compilation).beforeLoaders.tap(
+        ConsoleRemotePlugin.name,
+        (loaders, normalModule) => {
+          const { userRequest } = normalModule;
+
+          const moduleRequest = userRequest.substring(
+            userRequest.lastIndexOf('!') === -1 ? 0 : userRequest.lastIndexOf('!') + 1,
+          );
+
+          if (!modifiedModules.includes(moduleRequest) && transformImports(moduleRequest)) {
+            normalModule.loaders.push({
```

Comment:
This is where our custom loader gets added to modules that match the `transformImports` filter.

webpack loaders are applied in reverse order, so the last one runs first.

This code was inspired https://github.com/artembatura/modify-source-webpack-plugin